Makefile

```diff
@@ -278,7 +279,11 @@ temporal-sql-tool: $(ALL_SRC)
 
 temporal-server-debug: $(ALL_SRC)
 	@printf $(COLOR) "Build temporal-server-debug with CGO_ENABLED=$(CGO_ENABLED) for $(GOOS)/$(GOARCH)..."
-	CGO_ENABLED=$(CGO_ENABLED) go build $(BUILD_TAG_FLAG),TEMPORAL_DEBUG -o temporal-server-debug ./cmd/server
+	CGO_ENABLED=$(CGO_ENABLED) go build $(BUILD_TAG_FLAG),with_assertions,TEMPORAL_DEBUG -o temporal-server-debug ./cmd/server
+
+temporal-server-dev: $(ALL_SRC)
+	@printf $(COLOR) "Build temporal-server-dev with assertions, with CGO_ENABLED=$(CGO_ENABLED) for $(GOOS)/$(GOARCH)..."
+	CGO_ENABLED=$(CGO_ENABLED) go build $(BUILD_TAG_FLAG),with_assertions -o temporal-server-dev ./cmd/server
```

Comment:
Having a separate "dev" build (a) makes every developer run Temporal with assertions and (b) avoids leaking the assertion build into prod accidentally.